fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/AbstractCumulativeLoanScheduleGenerator.java

```diff
@@ -410,6 +410,10 @@ private LoanScheduleModel generate(final MathContext mc, final LoanApplicationTe
             scheduleParams.addTotalCumulativeInterest(totalInterest);
         }
 
+        // add charges only repayment schedule for charges with due date after maturity date (N+1)
+        addChargesOnlyRepaymentSchedule(currency, scheduleParams, periods, loanCharges, mc, loanApplicationTerms, transactions,
```

Comment:
Is this only needed for cumulative loans? What about progressive ones?

Reply:
It is required for progressive ones also, but since interest handling, reschedule, prepayment  and related stories for progressive schedule handling are pending i did not changed for progressive.

Reply:
does that mean if a progressive N+1 loan got rescheduled the N+1 charge will disappear?

Reply:
I will test and confirm.

Reply:
@adamsaghy, verified with latest, for progressive loans also N+1 installment vanishes after reschedule.